### PR TITLE
Fix tests.

### LINKED ISSUE #1872: Fail to build with rustc 1.20.0-nightly (2fbba5bdb 2017-07-04)
Here's the [detail log](https://gist.github.com/uHOOCCOOHu/dd0deb6de4d164939ee41fac5311201b).
Seems caused by rust-lang/rust#43012.

### PR DESCRIPTION
**Pull request by uHOOCCOOHu**
Fix #1872.

### CODE REVIEW

File: clippy_tests/examples/range.rs [hunk 1]
```diff
@@ -29,7 +21,4 @@ fn main() {
     let v2 = vec![4,5];
     let _x = v1.iter().zip(0..v1.len());
     let _y = v1.iter().zip(0..v2.len()); // No error
-
-    // check const eval
-    let _ = v1.iter().step_by(2/3);
```
Comment (oli-obk) [inline clippy_tests/examples/range.rs]: You should leave these in, since they use the Iterator::step_by feature

Comment (oli-obk) [inline clippy_tests/examples/range.rs]: Sorry, that wasn't very clear... I meant to leave all the calls to step_by in the tests. At least those that compile.

Comment (uHOOCCOOHu) [inline clippy_tests/examples/range.rs]: I want to do so, but directly passing -1 or 0 to `step_by` causes a compile error immediately, which will suppress the clippy warnings. I don't think that's what we expected.

Comment (uHOOCCOOHu) [inline clippy_tests/examples/range.rs]: I'm sorry for it. `step_by(0)` in tests is ok. I'll put them back.

**Comment by oli-obk** [comment]
Great! Thanks.

**Pull request merged**